CIP-0071/README.md

```diff
@@ -0,0 +1,246 @@
+	* Mark their most recent vote to match the `vote` object in the UTxOs datum
+3. Ensure any required quorums or thresholds were reached
+4. Report on the final ballot outcome
+
+Javascript-like pseudocode using the [Lucid library](https://github.com/spacebudz/lucid) for the above algorithm would be as follows:
+
+```js
+function countVotes(ballotPolicyId, ballotBox) {
+  var votesByAsset = {};
+  const votes = await lucid.utxosAt(ballotBox);
+  for (const vote of votes) {
+    const voteResult = Data.toJson(Data.from(vote.datum));
+    for (const unit in vote.assets) {
+      if (!unit.startsWith(ballotPolicyId)) {
+        continue;
+      }
+      const voteCount = Number(vote.assets[unit]);         // Should always be 1
+      votesByAsset[unit] = {
+        voter: voteResult.voter,
+        vote: voteResult.vote,
+        count: voteCount
+      }
+    }
+  }
+  return votesByAsset;
+}
+```
+
+There is no requirement that the "ballot counter" redeem all "ballots" from the "ballot box" and send them back to the respective voters, but we anticipate that this is what will happen in practice.  We encourage further open-sourced code versions that enforce this requirement at the smart contract level.
+
+### Reclaiming ADA Locked by the Ballot NFTs
+
+Even if the ballot NFT is returned to the user, this will leave users with ADA locked alongside these newly created assets, which can impose a financial hardship for certain project users.
+
+We can add burn-specific code to our Plutus minting policy so that ballot creation does not impose a major financial burden on users:
+
+```ts
+func main(redeemer: Redeemer, ctx: ScriptContext) -> Bool {
+  tx: Tx = ctx.tx;
+  minted_policy: MintingPolicyHash = ctx.get_current_minting_policy_hash();
+  redeemer.switch {
+    // Minting code elided for space...
+    Burn => {
+      tx.minted.get_policy(minted_policy).all((asset_id: ByteArray, amount: Int) -> Bool {
+        if (amount > 0) {
+          print(asset_id.show() + " asset ID was minted not burned (quantity " + amount.show() + ")");
+          false
+        } else {
+          true
+        }
+      })
+    }
+  }
+}
+```
+
+The Helios code above simply checks that during a burn (as indicated by the Plutus minting policy's `redeemer`), the user is not attempting to mint a positive number of any assets.  With this code, *any Cardano wallet* can burn *any ballot* minted as part of this protocol.  Why so permissive? We want to ensure that each vote is bringing the minimal costs possible to the user.  In providing this native burning mechanism we can free up the minUTxO that had been locked with the ballot, and enable the user to potentially participate in more votes they might not have otherwise.  In addition, users who really do not like the specific commemorative NFTs or projects that choose to skip the "commemorative" aspect of ballot creation now have an easy way to dispose of "junk" assets.
+
+## Potential Attacks and Mitigations
```

Comment:
Please nest this under a `Rationale` section.

Reply:
Done.